src/App.tsx

```diff
@@ -8,37 +8,88 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getPostsData, getUsersData } from './api/posts';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [loader, setLoader] = useState<boolean>(false);
```

Comment:
Rename all boolean variables using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(false);
```